streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignorTest.java

```diff
@@ -997,6 +997,77 @@ public void testAssignWithStandbyReplicas() {
         assertEquals(standbyPartitionsByHost, info20.standbyPartitionByHost());
     }
 
+    @Test
+    public void testAssignWithStandbyReplicasBalance() {
```

Comment:
Could we add some tests to cover the are less than total tasks (active + standby), but we can still distribute them evenly, ex:
tasks: TASK_0_0, TASK_0_1, TASK_0_2
consumers: 2 in thread1, 3 in thread2
We expected each thread should have 3 tasks, and distribute evenly.

Reply:
Also, some tests to make sure the `statelessTasks` are also taking the thread load into account. Thanks.